src/main/java/storage/FileManager.java

```diff
@@ -0,0 +1,244 @@
+        if (!isWasDirectoryAlreadyMade()) {
+            ui.printDirectoryNotFoundMessage();
+            createDataDirectory();
+            ui.printDirectoryCreatedMessage(getDirectoryPath());
+        }
+
+        checkIfExerciseFileAlreadyExists();
+        if (!isWasExercisesFileAlreadyMade()) {
+            ui.printExerciseFileNotFoundMessage();
+            createExerciseFile();
+            ui.printExerciseFileCreatedMessage(getExerciseFilePath());
+        }
+
+        checkIfWorkoutFileAlreadyExists();
+        if (!isWasWorkoutsFileAlreadyMade()) {
+            ui.printWorkoutFileNotFoundMessage();
+            createWorkoutFile();
+            ui.printWorkoutFileCreatedMessage(getWorkoutFilePath());
+        }
+    }
+
+    public void checkIfDataDirectoryAlreadyExists() {
+        if (Files.exists(getDirectoryPath())) {
+            setWasDirectoryAlreadyMade(true);
+        }
+    }
+
+    public void createDataDirectory() throws IOException {
+        Files.createDirectory(getDirectoryPath());
+    }
+
+    public void checkIfExerciseFileAlreadyExists() {
+        if (Files.exists(getExerciseFilePath())) {
+            setWasExercisesFileAlreadyMade(true);
+        }
+    }
+
+    public void createExerciseFile() throws IOException {
+        Files.createFile(getExerciseFilePath());
+
+        // Populate file with default exercises
+        FileWriter fileWriter = new FileWriter(getExerciseFilePath().toString());
+        for (String exerciseName : ExerciseList.getDefaultExerciseList()) {
+            fileWriter.append(exerciseName);
+            fileWriter.append(System.lineSeparator());
+        }
+        fileWriter.close();
+    }
+
+    public void checkIfWorkoutFileAlreadyExists() {
+        if (Files.exists(getWorkoutFilePath())) {
+            setWasWorkoutsFileAlreadyMade(true);
+        }
+    }
+
+    public void createWorkoutFile() throws IOException {
+        Files.createFile(getWorkoutFilePath());
+    }
+
+    /**
+     * Reads the exercises from local file and store them into an ArrayList named exercises.
+     *
+     * @throws IOException when the local file is missing.
+     */
+    public void loadExercisesFromFile(ExerciseList exerciseList) throws IOException {
+        Scanner scanner = new Scanner(getExerciseFilePath());
+        while (scanner.hasNextLine()) {
+            var exercise = scanner.nextLine();
+            exerciseList.addExerciseToList(exercise);
+        }
+    }
+
+    public boolean loadWorkoutsFromFile(WorkoutList workoutList) throws IOException {
+        boolean hasNoErrorsDuringLoad = true;
+        Scanner workoutFileReader = new Scanner(getWorkoutFilePath());
+        while (workoutFileReader.hasNext()) {
+            try {
+                String workoutFileDataLine = workoutFileReader.nextLine();
+                String[] parsedWorkoutFileDataLine = parseFileDataLine(workoutFileDataLine);
+                addFileWorkoutToList(workoutList, parsedWorkoutFileDataLine);
+
```

Comment:
Minor formatting issue - just remove space!

Reply:
Thanks for spotting this. Removed in Commit ce06262